src/axolotl/cli/merge_lora.py

```
@@ -1,21 +1,67 @@
-"""
-CLI to run merge a trained LoRA into a base model
-"""
+"""CLI to merge a trained LoRA into a base model."""
+
+import logging
 from pathlib import Path
 from typing import Union
 
 import fire
 import transformers
 from dotenv import load_dotenv
 
-from axolotl.cli import do_merge_lora, load_cfg, print_axolotl_text_art
-from axolotl.common.cli import TrainerCliArgs
+from axolotl.cli.art import print_axolotl_text_art
+from axolotl.cli.config import load_cfg
+from axolotl.common.cli import TrainerCliArgs, load_model_and_tokenizer
+from axolotl.utils.dict import DictDefault
 
+LOG = logging.getLogger(__name__)
 
-def do_cli(config: Union[Path, str] = Path("examples/"), **kwargs):
-    # pylint: disable=duplicate-code
+
+def do_merge_lora(*, cfg: DictDefault) -> None:
+    """
+    Calls `transformers`' `merge_and_unload` on the model given in the `axolotl` config
+    along with the LoRA adapters to combine them into a single base model.
+
+    Args:
+        cfg: Dictionary mapping `axolotl` config keys to values.
+    """
     print_axolotl_text_art()
-    parser = transformers.HfArgumentParser((TrainerCliArgs))
+
+    model, tokenizer = load_model_and_tokenizer(cfg=cfg)
+    safe_serialization = cfg.save_safetensors is True
+
+    LOG.info("Running merge of LoRA with base model...")
+    model = model.merge_and_unload(progressbar=True)
+    try:
+        model.to(dtype=cfg.torch_dtype)
+    except RuntimeError:
+        pass
```

How come we're failing silently when the user attempts to cast to a datatype that would error out?

---

No idea, this code is inherited. @winglian @NanoCode012 can we remove?